From a982cfb5950b92c1856fcda3dba8ab5e26c56f78 Mon Sep 17 00:00:00 2001
From: Alexander Filonenko <vtp684@motorola.com>
Date: Thu, 30 May 2013 14:25:20 -0500
Subject: [PATCH 044/959] IKJBDVX-391 misc: utags: get blksize from inode

Fetch blksize from inode directly instead of ioctl call.

Signed-off-by: Alexander Filonenko <vtp684@motorola.com>

Change-Id: I4fd0bfa1346effdc0d7bebcd90de346bc34fa50f
Reviewed-on: http://gerrit.pcs.mot.com/544785
SLT-Approved: Slta Waiver <sltawvr@motorola.com>
Tested-by: Jira Key <jirakey@motorola.com>
Reviewed-by: Dmitry Ornatsky <cdo021c@motorola.com>
Reviewed-by: Alexander Filonenko <vtp684@motorola.com>
Reviewed-by: Jason Hrycay <jason.hrycay@motorola.com>
Submit-Approved: Jira Key <jirakey@motorola.com>
---
 drivers/misc/utag/utags.c | 19 ++-----------------
 1 file changed, 2 insertions(+), 17 deletions(-)

diff --git a/drivers/misc/utag/utags.c b/drivers/misc/utag/utags.c
index fcc431a0e13..74e6ce8ac80 100644
--- a/drivers/misc/utag/utags.c
+++ b/drivers/misc/utag/utags.c
@@ -469,7 +469,6 @@ static void *freeze_tags(size_t block_size, const struct utag *tags,
  */
 static struct utag *load_utags(const char *partition_name)
 {
-	int res;
 	size_t block_size;
 	ssize_t bytes;
 	struct utag *head = NULL;
@@ -499,14 +498,7 @@ static struct utag *load_utags(const char *partition_name)
 		goto close_block;
 	}
 
-	res =
-	    filp->f_op->unlocked_ioctl(filp, BLKGETSIZE64,
-				       (unsigned long)&block_size);
-	if (0 > res) {
-		pr_err("%s ERR file (%s) ioctl failed\n", __func__,
-		       partition_name);
-		goto close_block;
-	}
+	block_size = i_size_read(inode->i_bdev->bd_inode);
 
 	data = kmalloc(block_size, GFP_KERNEL);
 	if (!data) {
@@ -566,7 +558,6 @@ replace_first_utag(struct utag *head, const char *name, void *payload,
 static enum utag_error
 flash_partition(const char *partition_name, const struct utag *tags)
 {
-	int res;
 	size_t written = 0;
 	size_t block_size = 0, tags_size = 0;
 	char *datap = NULL;
@@ -595,13 +586,7 @@ flash_partition(const char *partition_name, const struct utag *tags)
 		goto close_block;
 	}
 
-	res =
-	    filep->f_op->unlocked_ioctl(filep, BLKGETSIZE64,
-					(unsigned long)&block_size);
-	if (0 > res) {
-		status = UTAG_ERR_PARTITION_NOT_BLKDEV;
-		goto close_block;
-	}
+	block_size = i_size_read(inode->i_bdev->bd_inode);
 
 	datap = freeze_tags(block_size, tags, &tags_size, &status);
 	if (!datap)
-- 
2.11.0

